debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java

```diff
@@ -408,6 +414,13 @@ protected void handleGtidEvent(Event event) {
         String gtid = gtidEvent.getGtid();
         gtidSet.add(gtid);
         source.startGtid(gtid, gtidSet.toString()); // rather than use the client's GTID set
+        ignoreDmlEventByGtidSource = false;
+        if (gtidDmlSourceFilter != null && gtid != null) {
+            String uuid = gtid.trim().substring(0, gtid.indexOf(":"));
+            if (!gtidDmlSourceFilter.test(uuid)) {
```

Comment:
Nice, that should work. According to this a UUID can only consist of alphnumerics and hyphens: https://docs.oracle.com/javase/7/docs/api/java/util/UUID.html#toString()